### PR TITLE
add --state option to the result method example

### PR DESCRIPTION
**Pull request by bsuryachandra**
## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `website/static/_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

**Comment by mirnawong1** [comment]
hey @bsuryachandra thanks for opening this pr! we'll have a look at this for you soon!

File: website/docs/reference/node-selection/methods.md [hunk 1]
```diff
@@ -227,10 +227,10 @@ $ dbt ls    --select +metric:* --resource-type source  # list all source tables
 The `result` method is related to the `state` method described above, and can be used to select resources based on their result status from a prior run. Note that one of the dbt commands [`run`, `test`, `build`, `seed`] must have been performed in order to create the result on which a result selector operates. You can use `result` selectors in conjunction with the `+` operator.
```
Comment (mirnawong1) [inline website/docs/reference/node-selection/methods.md]:
```suggestion
The `result` method is related to the `state` method described above and can be used to select resources based on their result status from a prior run. Note that one of the dbt commands [`run`, `test`, `build`, `seed`] must have been performed in order to create the result on which a result selector operates. You can use `result` selectors in conjunction with the `+` operator. 
```

File: website/docs/reference/node-selection/methods.md [hunk 2]
```diff
@@ -224,13 +224,13 @@ $ dbt ls    --select +metric:* --resource-type source  # list all source tables
...
 ```bash
-$ dbt run --select result:error # run all models that generated errors on the prior invocation of dbt run
-$ dbt test --select result:fail # run all tests that failed on the prior invocation of dbt test
-$ dbt build --select 1+result:fail # run all the models associated with failed tests from the prior invocation of dbt build
-$ dbt seed --select result:error # run all seeds that generated errors on the prior invocation of dbt seed.
+$ dbt run --select result:error --state path/to/artifacts # run all models that generated errors on the prior invocation of dbt run
+$ dbt test --select result:fail --state path/to/artifacts # run all tests that failed on the prior invocation of dbt test
+$ dbt build --select 1+result:fail --state path/to/artifacts # run all the models associated with failed tests from the prior invocation of dbt build
+$ dbt seed --select result:error --state path/to/artifacts # run all seeds that generated errors on the prior invocation of dbt seed.
 ```
```
Comment (mirnawong1) [inline website/docs/reference/node-selection/methods.md]:
```suggestion
**Note** &mdash; When you use dbt Cloud, please be aware that the `--state` parameter is automatically included when you click the [**Defer**](/docs/deploy/cloud-ci-job#deferral-and-state-comparison) button.
```

File: website/docs/reference/node-selection/methods.md [hunk 3]
```diff
@@ -224,15 +224,15 @@ $ dbt ls    --select +metric:* --resource-type source  # list all source tables
...
-
+**Note** &mdash; When you use dbt Cloud, please be aware that the `--state` parameter is automatically included when you click the [**Defer**](/docs/deploy/cloud-ci-job#deferral-and-state-comparison) button.
```
Comment (b-per) [inline website/docs/reference/node-selection/methods.md]:
```suggestion
**Note** &mdash; When you define steps in your dbt Cloud jobs, please be aware that the `--state` parameter is automatically included when you click the [**Defer**](/docs/deploy/cloud-ci-job#deferral-and-state-comparison) button.
```

Comment (b-per) [reply website/docs/reference/node-selection/methods.md]: The only slight clarification is that `--state` is automatically added for **job steps** when the deferral box is ticked.

In the IDE for example people would still have to type `--state ...`

**Review by b-per — DISMISSED** [review]
Minor change

**Comment by mirnawong1** [comment]
thanks for that clarification @b-per !

**Comment by mirnawong1** [comment]
hey @bsuryachandra, thanks again for opening this pr and contributing to the docs ✨ with Benoit's help, i made some small tweaks to further clarify. going to merge this now and looking forward to future contributions!

**Review by mirnawong1 — APPROVED** [review]

**Comment by bsuryachandra** [comment]
Thanks for your help @mirnawong1

**Pull request merged**